src/Migrations/M032MigrateCampaignCoversToP4ColumnsBlock.php

```diff
@@ -0,0 +1,165 @@
+    {
+        try {
+            // Get the list of posts using Covers block.
+            $posts = Utils\Functions::get_posts_using_specific_block(
+                Utils\Constants::BLOCK_COVERS,
+                Utils\Constants::ALL_POST_TYPES
+            );
+
+            // If there are no posts, abort.
+            if (!$posts) {
+                return;
+            }
+
+            $parser = new WP_Block_Parser();
+
+            echo "Campaign Covers block migration in progress...\n"; // phpcs:ignore
+
+            foreach ($posts as $post) {
+                if (empty($post->post_content)) {
+                    continue;
+                }
+
+                $current_post_id = $post->ID; // Store the current post ID
+
+                // Go through the post blocks to find the planet4-blocks/covers one.
+                $blocks = $parser->parse($post->post_content);
+
+                foreach ($blocks as &$block) {
+                    // Skip non cover block.
+                    if (!isset($block['blockName']) || $block['blockName'] !== Utils\Constants::BLOCK_COVERS) {
+                        continue;
+                    }
+
+                    // For older cover blocks where the cover type is empty,
+                    // the default content cover type is applied to those blocks.
+                    if (!isset($block['attrs']['cover_type'])) {
+                        continue;
+                    }
+
+                    // For old cover type(earlier it was numeric).
+                    if (is_numeric($block['attrs']['cover_type'])) {
+                        // phpcs:ignore Generic.Files.LineLength.MaxExceeded
+                        $block['attrs']['cover_type'] = Utils\Constants::OLD_COVER_TYPES[ $block['attrs']['cover_type'] ];
+                    }
+
+                    // Skip non campaign cover style blocks.
+                    if ($block['attrs']['cover_type'] !== Utils\Constants::COVER_TYPE_CAMPAIGN) {
+                        continue;
+                    }
+
+                    $block = self::transform_block($block);
+                }
+
+                // Unset the reference to prevent potential issues.
+                unset($block);
+
+                // Serialize the blocks content.
+                $new_content = serialize_blocks($blocks);
+
+                if ($post->post_content === $new_content) {
+                    continue;
+                }
+
+                echo 'Migrating post ', $post->ID, "\n"; // phpcs:ignore
+
+                $post_update = array(
+                    'ID' => $post->ID,
+                    'post_content' => $new_content,
+                );
+
+                // Update the post with the replaced blocks.
+                $result = wp_update_post(wp_slash($post_update));
+
+                if ($result === 0) {
+                    throw new \Exception("There was an error trying to update the post #" . $post->ID);
```

Comment:
Same question here, we could use `$current_post_id`

Reply:
It was a miss, Thanks heads up 🙏